frontend/server/src/Controllers/QualityNomination.php

```diff
@@ -71,6 +73,9 @@ public static function apiCreate(\OmegaUp\Request $r) {
         \OmegaUp\Validators::validateStringNonEmpty($r['problem_alias'], 'problem_alias');
         \OmegaUp\Validators::validateInEnum($r['nomination'], 'nomination', ['suggestion', 'promotion', 'demotion', 'dismissal']);
         \OmegaUp\Validators::validateStringNonEmpty($r['contents'], 'contents');
+        /**
+         * @var null|array{tags: mixed} $contents
+         */
```

Comment:
```suggestion
        /** @var null|array{tags?: mixed, before_ac?: mixed, difficulty?: mixed, quality?: mixed, ...} $contents */
```

aquí hay que enumerar todas las cosas que el arreglo puede contener. normalmente hay que evitar el uso de `mixed` de ser posible, pero como esto nos lo está mandando un usuario que _podría_ mandarnos cualquier basura que quiera, aquí sí se vale usar `mixed` (y realmente no podemos usar ningún otro tipo). por eso todos esos `isset(...) && is_int(...)` y así.